src/main/java/com/facebook/openwifirrm/modules/ProvMonitor.java

```diff
@@ -105,29 +110,58 @@ private void runImpl() {
 		// ignoring "entity" completely
 		InventoryTagList inventory = client.getProvInventory();
 		SerialNumberList inventoryForRRM = client.getProvInventoryForRRM();
-		VenueList venueList = client.getProvVenues();
 		//EntityList entityList = client.getProvEntities();
 		if (inventory == null || inventoryForRRM == null) {
 			logger.error("Failed to fetch inventory from owprov");
 			return;
 		}
+
+		// fetch the RRM details for each venue
+		Map<String, RRMDetails> rrmDetails = new HashMap<String, RRMDetails>();
+		// TODO this currently chooses the first AP encountered per venue to fetch
+		// the RRMDetails. Once a proper venue based RRM settings API is available
+		// we should use that instead of doing this.
+		for (InventoryTag tag : inventory.taglist) {
+			rrmDetails.computeIfAbsent(tag.venue, k -> {
```

Comment:
Skip if `tag.venue.isEmpty()` (a device may not have a venue if it got auto-added but not yet configured)